src/main/java/sirius/biz/util/ExtractedFileBuffer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.util;
+
+import org.apache.commons.io.output.DeferredFileOutputStream;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.Log;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Wraps a {@link DeferredFileOutputStream stream} that contains an extracted file either retained in memory or in
+ * a temporary file.
+ */
+public class ExtractedFileBuffer {
```

Comment:
implement Closable?

Reply:
I thought about that, but we only close the output stream but no input streams. I'm not sure if that would confuse people, but i can change that if you'd like.

Reply:
make this class package private then (?)

Reply:
Done